website/app/lib/events-air.server.ts

```diff
@@ -0,0 +1,393 @@
+import { trace } from '@opentelemetry/api'
+import { EVENTS_AIR_CLIENT_ID, EVENTS_AIR_CLIENT_SECRET, EVENTS_AIR_TENANT_ID } from './config.server'
+
+// EventsAir contact data type
+export interface EventsAirContactData {
+    firstName: string | null
+    lastName: string | null
+    primaryEmail: string | null
+    externalIdentifier: string | null
+    userDefinedField1?: string
+    userDefinedField2?: string
+    userDefinedField3?: string
+}
+
+export async function getAccessToken(): Promise<string> {
+    if (!EVENTS_AIR_CLIENT_ID || !EVENTS_AIR_CLIENT_SECRET) {
+        throw new Error('EventsAir client ID and secret are not configured')
+    }
+
+    return await trace.getTracer('default').startActiveSpan('getAccessToken', async (span) => {
+        try {
+            const tokenResponse = await fetch(
+                `https://login.microsoftonline.com/${EVENTS_AIR_TENANT_ID}/oauth2/v2.0/token`,
+                {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/x-www-form-urlencoded',
+                    },
+                    body: new URLSearchParams({
+                        client_id: EVENTS_AIR_CLIENT_ID!,
+                        client_secret: EVENTS_AIR_CLIENT_SECRET!,
+                        scope: 'https://eventsairprod.onmicrosoft.com/85d8f626-4e3d-4357-89c6-327d4e6d3d93/.default',
```

Comment:
should this scope be using the EVENTS_AIR_TENANT_ID ? Or the client id? I guess it looks like magic here

Reply:
(85d8f626-4e3d-4357-89c6-327d4e6d3d93 that is)

Reply:
It's a fixed scope